runtime/extern.c

```diff
@@ -745,6 +777,27 @@ static void extern_rec(value v)
       extern_record_location(v, h);
       break;
     }
+#ifdef NO_NAKED_POINTERS
+    case Closure_tag: {
+      mlsize_t i;
+      extern_header(sz, tag);
+      size_32 += 1 + sz;
+      size_64 += 1 + sz;
+      extern_record_location(v, h);
+      i = extern_closure_up_to_env(v);
+      if (i >= sz) goto next_item;
+      /* Remember that we still have to serialize fields i + 1 ... sz - 1 */
+      if (sz - i > 1) {
```

Comment:
```suggestion
      if (i < sz - 1) {
```
I'd like this test to be more clearly related to the previous one (two lines up).